checkstyle-tester/diff.groovy

```diff
@@ -0,0 +1,232 @@
+import java.nio.file.Paths
+
+import static java.lang.System.err
+
+def cli = new CliBuilder(usage:'groovy diff.groovy [options]')
+cli.with {
+    r(longOpt: 'localGitRepo', args: 1, argName: 'localGitRepo', 'Path to local git repository')
+    b(longOpt: 'patchBranch', args: 1, argName: 'patchBranch', 'Name of the patch branch in local git repository')
+    c(longOpt: 'checkstyleCfg', args: 1, argName: 'checkstyleCfg', 'Path to checkstyle config file')
+    p(longOpt: 'projectsToTestOn', args: 1, argName: 'projectsToTestOn', 'Path to file which contains projects to test on')
+}
+def options = cli.parse(args)
+
+if (areValidCliArgs(options)) {
+    def localGitRepo = new File(options.localGitRepo)
+    if (hasUnstagedChanges(localGitRepo)) {
+        throw new IllegalStateException("Error: git repository ${localGitRepo.getPath()} has unstaged changes!")
+    }
+
+    def patchBranch = options.patchBranch
+    def projectsToTestOn = options.projectsToTestOn
+    def checkstyleCfg = options.checkstyleCfg
+
+    def reportsDir = 'reports'
+    def reportsBeforeFixDir = "$reportsDir/before_fix"
+    def reportsAfterFixDir = "$reportsDir/after_fix"
```

Comment:
There should be no term "fix".
It is just comparison tool of two report that are done on some branches.
so name them as it is - "checkstyle-checkstyle-master" vs "MEZk-checkstyle-i3333".
In terms of this script: localGitRepo v patchBranch ..... master vs i3333